src/execution.cc

```diff
@@ -51,7 +51,73 @@ int GuessParallelism() {
   }
 }
 
+Node* TargetNameToNode(const State* state, const std::string& path, std::string* err) {
+  uint64_t slash_bits;
+  std::string canonical_path = path.c_str();
```

Comment:
This requires that the constructor for canonical_path run strlen on the provided char*.

Better to simply do

```cpp
std::string canonical_path = path;
```